tests/registration_test.py

```diff
@@ -306,6 +306,44 @@ def test_string_is_registered(self):
         self.assertFalse(keep.is_registered())
         ak.clear()
 
+    def test_delete_registered(self):
+        """
+        Tests the following:
+
+        1. delete cmd doesn't delete registered objects and returns appropriate message
+        2. delete cmd does delete non-registered objects and returns appropriate message
+        3. delete cmd raises RuntimeError for unknown symbols
+        """
+        cleanup()
+        a = ak.ones(3, dtype=ak.int64)
+        b = ak.ones(3, dtype=ak.int64)
+
+        # registered objects are not deleted from symbol table
+        a.register('keep')
+        self.assertEqual(ak.client.generic_msg(cmd='delete', args='keep'),
+                         'failed to delete registered symbol keep')
+        self.assertTrue(a.name in list_symbol_table())
+
+        # non-registered objects are deleted from symbol table
+        self.assertEqual(ak.client.generic_msg(cmd='delete', args=b.name),
+                         'deleted ' + b.name)
+        self.assertTrue(b.name not in list_symbol_table())
+
+        # RuntimeError when calling delete on an object not in the symbol table
+        with self.assertRaises(RuntimeError):
+            ak.client.generic_msg(cmd='delete', args='not_in_table')
+
+def list_symbol_table() -> List[str]:
+    symbol_list: List[str] = []
+
+    if ak.info(ak.AllSymbols) not in [ak.EmptyRegistry, ak.EmptySymbolTable]:
+        bject: str
```

Comment:
this is a typo, should be `object: str`. I'm a bit surprised this didn't cause issues. I'll fix this along with @hokiegeek2s comments